lib/util/http_certificate_check.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/service/remote_config_service.dart';
+import 'package:autonomy_flutter/util/bytes_utils.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:sentry/sentry.dart';
+
+Future<String?> _fetchCertificate(Uri url) async {
+  try {
+    // Create an HttpClient to make a request and get the certificate
+    final client = HttpClient();
+    final request = await client.getUrl(url);
```

Comment:
```suggestion
    final request = await client.headUrl(url);
```

Please try with HEAD method since we don't need to get the content.